src/gedi_subset/subset.py

```diff
@@ -256,6 +279,14 @@ def main(
     lon: str = typer.Option(
         ..., help=("Longitude dataset used in the geometry of the dataframe")
     ),
+    beams: str = typer.Option(
+        "all",
```

Comment:
weird, looks like my original comment was lost, or I accidentally closed before submitting it

add:

```
        callback=check_beams_option,
```

and then put this function somewhere:

```python
def check_beams_option(value: str) -> str | NoReturn:
    upper_value = value.upper()
    suffixes = [name.strip().lstrip("BEAM") for name in upper_value.split(",")]
    valid_suffixes = ["0000", "0001", "0010", "0011", "0101", "0110", "1000", "1011"]

    if upper_value not in ["ALL", "COVERAGE", "POWER"] and any(
        suffix not in valid_suffixes for suffix in suffixes
    ):
        raise BadParameter(value)

    return (
        ",".join(f"BEAM{suffix}" for suffix in suffixes)
        if len(suffixes) > 1
        else upper_value
    )
```